### ISSUE TITLE
Pdf is not loading in fragment but it is working in activity

### ISSUE THREAD

**Issue by asheshchalise** [barteksc/AndroidPdfViewer#862]

**Comment by realityworks** [comment]
I have the same problem here. Any ideas? Version 2.8.0

**Comment by 1stmetro** [comment]
have you checked your using the correct context ?

**Comment by 1stmetro** [comment]
I presume your pdf code is in the fragment not in the main activity ?

**Comment by realityworks** [comment]
Yes, the PDF code is in the fragment like this :

```
if (file.exists()) {
        lesson.fromFile(file)
            .enableSwipe(true)
            .swipeHorizontal(false)
            .onLoad(object : OnLoadCompleteListener {
              override fun loadComplete(l: Int) {
              }
            }).onPageChange { page, pageCount ->
              if (page == pageCount) {
                val prefKey = "L" + lessonNumber + "_" + name
                savePref(prefKey)
              }
            }
            .load()
        lesson.visibility = View.VISIBLE
    } else {
        throw FileNotFoundException()
    }
  }
```

Gets called from onStart

@1stmetro

**Comment by 1stmetro** [comment]
whats the error ?

**Comment by realityworks** [comment]
It just shows a white screen. No PDF.

**CONSOLE**
D/LessonsDetailFragment: Current lesson : 3 and current groupNumber : 1
I/Choreographer: Skipped 610 frames!  The application may be doing too much work on its main thread.
I/OpenGLRenderer: Davey! duration=10326ms; Flags=0, IntendedVsync=22421526043161, Vsync=22431692709421, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=22431709957767, AnimationStart=22431710911767, PerformTraversalsStart=22431715144767, DrawStart=22431823509767, SyncQueued=22431843995767, SyncStart=22431844165767, IssueDrawCommandsStart=22431844446767, SwapBuffers=22431845622767, FrameCompleted=22431852677767, DequeueBufferDuration=55000, QueueBufferDuration=3586000, 
D/jniPdfium: Init FPDF library

**Comment by realityworks** [comment]
Then that's it.

**Comment by realityworks** [comment]
When run in an activity I see this in the console.

**CONSOLE**
D/jniPdfium: Init FPDF library
D/EGL_emulation: eglMakeCurrent: 0xcc4579e0: ver 3 0 (tinfo 0xca97e460)
I/chatty: uid=10093(org.bwf.shuttletime.android) RenderThread identical 2 lines
D/EGL_emulation: eglMakeCurrent: 0xcc4579e0: ver 3 0 (tinfo 0xca97e460)
I/tletime.androi: NativeAlloc concurrent copying GC freed 11340(551KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 3MB/6MB, paused 6.714ms total 66.898ms

**Comment by realityworks** [comment]
Is this a context related issue?

**Comment by 1stmetro** [comment]
It might be worth trying a few pdf's but you must be doing something else due to the frame loss, put the pdf code into a button for now to remove the lagging and the memory consumption.

**Comment by 1stmetro** [comment]
If it was a context issue i would expect to see a null error.

**Comment by realityworks** [comment]
@1stmetro where can I see reference code for putting it into a button? You mean in the Layout?

**Comment by 1stmetro** [comment]
Button myButton = (Button) findViewById(R.id.youButtonName);
myButton.setOnClickListener(new View.OnClickListener() {
    public void onClick(View v) {
        // Put your code that will run on a click here.
    }
});

**Comment by realityworks** [comment]
Yes, but our app starts downloading the PDF as soon as the screen opens. We then display is straight away.

**Comment by 1stmetro** [comment]
The downloading i'm guessing is causing the lag as its on your main thread, you have to ensure the file has downloaded before you try to open it otherwise i guess it would be a blank page.

**Comment by realityworks** [comment]
I check to see if the saved file exists, if it does, I show it. If not, I throw an exception and catch it, download the file and show it again.

The file exists when I try to show it, so I don't think it is that.

Also, I do the same thing in an activity and it works fine on that. 

Here is the activity code 👍 
```
        if (file.exists()) {
            pdfView.fromFile(file)
                //.pages(*getIntent().getIntArrayExtra("pages"))
                .enableDoubletap(true)
                .defaultPage(2)
                .swipeHorizontal(false)
                .enableSwipe(true)
                .onPageChange(object : OnPageChangeListener {
                    override fun onPageChanged(page: Int, pageCount: Int) {
                        toolbar_title.setText(title)
                    }
                })
                .load()
            pdfView.visibility = View.VISIBLE
        } else {
            error_text_view.visibility = View.VISIBLE
            pdfView.visibility = View.GONE
        }
        pdf_loading.setVisibility(View.GONE)
```

FRAGMENT CODE
```
    if (file.exists()) {
        lesson.fromFile(file)
            .enableSwipe(true)
            .swipeHorizontal(false)
            .onLoad(object : OnLoadCompleteListener {
              override fun loadComplete(l: Int) {
              }
            }).onPageChange { page, pageCount ->
              if (page == pageCount) {
                val prefKey = "L" + lessonNumber + "_" + name
                savePref(prefKey)
              }
            }
            .load()
        lesson.visibility = View.VISIBLE
    } else {
        throw FileNotFoundException()
    }
```

**Comment by realityworks** [comment]
Also, seems like 

```
.onRender(object : OnRenderListener {
                    override fun onInitiallyRendered(nbPages: Int, pageWidth: Float, pageHeight: Float) {
                        Log.d("LessonsDetailFragment", "On initially rendered")
                    }
                })
```

On the fragment code is never called...

The Activity code has the log message "On Initially Rendered". But when I run the same code in a fragment, it doesn't work.

**Comment by realityworks** [comment]
@1stmetro 

Stepping through while debugging, the problem is that the following call is never made in the code from the fragment.

```
    private void drawWithListener(Canvas canvas, int page, OnDrawListener listener) {
```

But the activity does break on that function call.

**Comment by realityworks** [comment]
I figured out the problem.

The frame layout of the parent view is returning a width and height of 0. So the PDFView is never rendered.

If I set the width and height manually everything works as expected! 

Thanks for your help and I believe we can close this issue, it's more of a layout problem than an actual issue with your library!

@1stmetro
